Ginger/GingerCoreNET/ALMLib/Azure/AzureDevOpsCore.cs

```diff
@@ -318,29 +308,73 @@ public bool ExportBusinessFlow(BusinessFlow businessFlow, ALMTestSetData mappedT
                 return false;
             }
         }
-        public int? CreateNewTestSuite(BusinessFlow bf, string fatherId, ObservableList<ExternalItemFieldBase> testsetfield)
+
+        public  void TestCaseEntryInSuite(BusinessFlow bf)
         {
-            LoginDTO logincred = GetLoginDTO();
+            try
+            {
+
+            
+            LoginDTO login = GetLoginDTO();
+            string projectName = login.Project;
+            VssConnection connection = AzureDevOpsRepository.LoginAzure(login);
+            int planId = Int32.Parse(bf.ExternalID); 
+
+            int suiteId = planId +1; // the root suite id will always be +1 from plan id
+
             // Get a testplan client instance
-            VssConnection connection = AzureDevOpsRepository.LoginAzure(logincred);
-            WorkItemTrackingHttpClient workItemTrackingClient = connection.GetClient<WorkItemTrackingHttpClient>();
-            WorkItem workItem;
-            JsonPatchDocument patchDocument = new JsonPatchDocument();
 
+            TestPlanHttpClient testPlanClient = connection.GetClient<TestPlanHttpClient>();
 
-            patchDocument.Add(
-               new JsonPatchOperation()
-               {
-                   Operation = Operation.Add,
-                   Path = "/fields/System.Title",
-                   Value = bf.Name
-               });
 
-            workItem = workItemTrackingClient.CreateWorkItemAsync(patchDocument, logincred.Project, "Test Suite").Result;
-            bf.ExternalID = workItem.Id.ToString();
             
-            return workItem.Id;
+                foreach (ActivitiesGroup ag in bf.ActivitiesGroups)
+                {
+                    WorkItem2 testcasetoAdd = new()
+                    {
+                        Id = Int32.Parse(ag.ExternalID)
+                    };
+
+                    SuiteTestCaseCreateUpdateParameters parameters = new()
+                    {
+                        workItem = testcasetoAdd
+                    };
+
+                    IEnumerable<SuiteTestCaseCreateUpdateParameters> parametersCollection = new List<SuiteTestCaseCreateUpdateParameters> { parameters };
+
+
+                    testPlanClient.AddTestCasesToSuiteAsync(parametersCollection, projectName, planId, suiteId);
+                }
+            }
+            catch (Exception)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Error related to Suite case entry");
+            }
+
+        }
+        public static void CreateNewTestPlanOrSuite(BusinessFlow bf)
+        {
```

Comment:
Change method name toe CreateTestPlan if code inside this functions is doing the same

Reply:
ok